amundsen_application/static/js/components/DashboardPage/index.tsx

```diff
@@ -0,0 +1,264 @@
+import * as React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { RouteComponentProps } from 'react-router';
+import * as qs from 'simple-query-string';
+
+import AvatarLabel from 'components/common/AvatarLabel';
+import Breadcrumb from 'components/common/Breadcrumb';
+import BookmarkIcon from 'components/common/Bookmark/BookmarkIcon';
+import Flag from 'components/common/Flag';
+import LoadingSpinner from 'components/common/LoadingSpinner';
+import TabsComponent from 'components/common/TabsComponent';
+import { getDashboard } from 'ducks/dashboard/reducer';
+import { GetDashboardRequest } from 'ducks/dashboard/types';
+import { GlobalState } from 'ducks/rootReducer';
+import { logClick } from 'ducks/utilMethods';
+import { Dashboard } from 'interfaces/Dashboard';
+import ImagePreview from './ImagePreview';
+import QueryList from 'components/DashboardPage/QueryList';
+import ChartList from 'components/DashboardPage/ChartList';
+import { formatDateTimeShort } from '../../utils/dateUtils';
+import ResourceList from 'components/common/ResourceList';
+import {
+  ADD_DESC_TEXT,
+  DASHBOARD_OWNER_SOURCE,
+  DASHBOARD_SOURCE,
+  LAST_RUN_SUCCEEDED,
+  NO_OWNER_TEXT,
+  TABLES_PER_PAGE
+} from 'components/DashboardPage/constants';
+import TagInput from 'components/Tags/TagInput';
+import { EditableSection } from 'components/TableDetail/EditableSection';
+import { ResourceType } from 'interfaces';
+
+import { getSourceDisplayName, getSourceIconClass } from 'config/config-utils';
+
+import './styles.scss';
+
+export interface RouteProps {
+  uri: string;
+}
+
+interface DashboardPageState {
+  uri: string;
+}
+
+export interface StateFromProps {
+  isLoading: boolean;
+  statusCode: number;
+  dashboard: Dashboard;
```

Comment:
nit: consider renaming `DashboardMetadata` to prevent confusion due to ambiguity.